File: src/matchMaker.Services/User/User.DataAccess/Dtos/ReportFilterDto.cs
using Common.Enums;

namespace User.DataAccess.Dtos;

public class ReportFilterDto
{
    public ReportStatus? Status { get; set; }
    public string? ReporterUserEmail { get; set; }
    public string? ReportedUserEmail { get; set; }
    public long? ReportTypeId { get; set; }
    public DateTime? CreatedFrom { get; set; }
    public DateTime? CreatedTo { get; set; }
    public bool? NotReviewed { get; set; }
}

File: src/ClientApp/src/app/dtos/report/ReportFilterDto.ts
import {ReportStatus} from "../../constants/report-status";

export interface ReportFilterDto {
  reporterUserEmail?: string;
  reportedUserEmail?: string;
  reportTypeId?: number;
  status?: ReportStatus;
  createdFrom?: string | null;
  createdTo?: string | null;
  notReviewed?: boolean;
}
